server/sonar-server/src/main/java/org/sonar/server/component/ComponentCleanerService.java

```diff
@@ -27,34 +27,23 @@
 import org.sonar.api.server.ServerSide;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
-import org.sonar.db.MyBatis;
 import org.sonar.db.component.ComponentDto;
-import org.sonar.server.component.index.ComponentIndexer;
-import org.sonar.server.issue.index.IssueIndexer;
-import org.sonar.server.measure.index.ProjectMeasuresIndexer;
-import org.sonar.server.test.index.TestIndexer;
+import org.sonar.server.es.ProjectIndexer;
+
+import static java.util.Arrays.asList;
 
 @ServerSide
 @ComputeEngineSide
 public class ComponentCleanerService {
 
   private final DbClient dbClient;
-  private final IssueIndexer issueIndexer;
-  private final TestIndexer testIndexer;
-  private final ProjectMeasuresIndexer projectMeasuresIndexer;
-  private final ComponentIndexer componentIndexer;
   private final ResourceTypes resourceTypes;
-  private final ComponentFinder componentFinder;
+  private final List<ProjectIndexer> projectIndexers;
 
-  public ComponentCleanerService(DbClient dbClient, IssueIndexer issueIndexer, TestIndexer testIndexer, ProjectMeasuresIndexer projectMeasuresIndexer,
-    ComponentIndexer componentIndexer, ResourceTypes resourceTypes, ComponentFinder componentFinder) {
+  public ComponentCleanerService(DbClient dbClient, ResourceTypes resourceTypes, ProjectIndexer[] projectIndexers) {
```

Comment:
Because of type erasure, it's not possible to declare pico dependencies through lists.